### DISCUSSION TITLE
Gerbera under Windows

### DISCUSSION THREAD

**Discussion by neheb** [gerbera/gerbera#1249] [Ideas]
Good or bad idea?

One interesting thing that breaks is fs::path. It returns std::wstring instead of std::string, causing all sorts of hell.

**Comment by whyman** [comment]
I think it would be a fun thing for someone to try and do, Im sure it would show up a whole load of issues.

**Comment by KarlStraussberger** [comment]
Very interesting, for sure. A lot more potential users.

There might be a couple more troublesome things, like inotify, which are not available and have to be reimplemented on basis of .NET Filesystem events. So, it's well worth considering.

**Reply by neheb** [reply]
Right. My short experiment with MinGW (which involved disabling almost everything) showed mainly the filesystem thing being problematic.

**Reply by neheb** [reply]
Current WIP: https://github.com/neheb/gerbera/commits/master

There are many assumptions across the codebase that fs::path is an std::string, which doesn't work.

**Reply by whyman** [reply]
Interesting, I think we should try to have UTF-8 strings only internally, conversion at the edges.

**Reply by neheb** [reply]
Rebased.

From what I see, several functions should be changed to handle an fs::path instead of std::string. It's an implicit conversion anyway.

**Reply by neheb** [reply]
UCRT might change it to std::string making the whole thing moot.

edit: just tested. still std::wstring.
